src/schema.jl

```diff
@@ -230,6 +232,63 @@ function apply_schema(t::ConstantTerm, schema::Schema, Mod::Type)
     InterceptTerm{t.n==1}()
 end
 
+# protection, borrowed from #117
+#
+# general idea is once we hit a FunctionTerm, we need to continue to
+# apply_schema because there might be some child that is un-protected.  So we
+# enter the Protected context and recursively apply_schema, and when we
+# encounter `unprotect` we restore the old context and continue to recursively
+# apply_schema
+struct Protected{Ctx} end
+Base.broadcastable(x::Protected) = Ref(x)
+# construct singletons to avoid method ambiguities using Type{<:Protected}
+protect(ctx) = Protected{ctx}()
+
+
+function apply_schema(t::FunctionTerm, schema::Schema, Mod::Type)
+    args = apply_schema.(t.args, schema, protect(Mod))
+    FunctionTerm(t.f, args, t.exorig)
+end
+
+apply_schema(t::FunctionTerm, schema::Schema, Ctx::Protected) =
+    FunctionTerm(t.f, apply_schema.(t.args, schema, Ctx), t.exorig)
+apply_schema(t, schema::Schema, Ctx::Protected) = t
+
+unprotect(t) = t
+unprotect(t::FunctionTerm{typeof(protect)}) = only(t.args)
+
+function apply_schema(t::FunctionTerm{typeof(unprotect)}, schema::Schema, Ctx::Protected{OldCtx}) where {OldCtx}
+    tt = only(t.args)
+    apply_schema(tt, schema, OldCtx)
+end
+
+
+# un-protecting special syntax: &, +, and *
+# 
+# if these occur as children of a non-special call, they are protected by
+# default.  so if we encounter them during apply_schema and we're NOT
+# in a Protected context, we need to call the corresponding op on the
+# # arguments of the FunctionCall
+
+# for op in (+, &, *)
```

Comment:
So others can use it.  But I'm not 100% convinced it'd actually be useful...

Reply:
Now that I've thought a bit more about this, I think it could be generally useful.  There's been some talk about developing better tooling for making extended syntax easier, and this seems like a step in that direction.